src/accessors.jl

```diff
@@ -0,0 +1,144 @@
+"""
+    get_datetimes(system)
+
+Extract datetimes from a `System`.
+"""
+function get_datetimes(system::System)
+    # use offer_curve axiskeys because all subtypes of System have offer_curve
+    return axiskeys(system.generator_time_series.offer_curve, 2)
+end
+
+get_zones(system::System) = system.zones
+
+"Returns a `Dictionary` with zonal regulation requirements indexed by zone number."
+function get_regulation_requirements(system::System)
+    return map(system.zones) do zone
+        zone.regulation
+    end
+end
+
+"Returns a `Dictionary` with zonal operating reserve requirements indexed by zone number."
+function get_operating_reserve_requirements(system::System)
+    return map(system.zones) do zone
+        zone.operating_reserve
+    end
+end
+
+"Returns a `Dictionary` with zonal good utility practice requirements indexed by zone number."
+function get_good_utility_requirements(system::System)
+    return map(system.zones) do zone
+        zone.good_utility
+    end
+end
+
+"Returns a `Dictionary` of `Bus` objects in the `System` indexed by bus name."
+get_buses(system::System) = system.buses
+"Returns a `Dictionary` of `Generator` objects in the `System` indexed by unit code."
+get_generators(system::System) = system.generators
+"Returns a `Dictionary` of `Branch` objects in the `System` indexed by branch name."
+get_branches(system::System) = system.branches
+
+"Returns a `Dictionary` of unit codes at each bus."
+get_gens_per_bus(system::System) = system.gens_per_bus
+"Returns a `Dictionary` of load names at each bus."
+get_loads_per_bus(system::System) = system.loads_per_bus
+
+"Returns the power transfer distribution factor of the system."
+get_ptdf(system::System) = system.PTDF
```

Comment:
Unrelated to this PR but this made me realise that PTDF and LODF are in upper case even though they're fields, not types. Shouldn't they be in lower case?

Reply:
Good point, changed to lower case.